acp/main_module.php

```diff
@@ -155,6 +157,20 @@ protected function get_sites()
 			];
 		}
 
+		$customSites = new CustomSitesCollection;
```

Comment:
Why not merge this array with the previous one, instead of duplicating things?

Reply:
`customsitescollection`

Reply:
> Why not merge this array with the previous one, instead of duplicating things?

This is because the ```$configurator->MediaEmbed->defaultSites``` is an object.

Reply:
I I noticed that... Why's it have to be such a complicated API. The only other thing I could thing of was moving that foreach loop into its own separate method.

Reply:
... or make the ```customsitescollection``` class implementing ```ArrayAccess ``` interface, but that would be another useless complication as for me.

Reply:
If the goal is to expand or override the list of default sites you can overwrite the existing definitions, e.g.

```php
$configurator->MediaEmbed->defaultSites->add(
	'ok',
	[
		'host'    => 'ok.ru',
		'name'    => 'Odnoklassniki',
		'extract' => [
			'!ok.ru/video/(?<id>\\d+)!',
			'!ok.ru/live/(?<id>\\d+)!',
		],
		'iframe'  => ['src' => 'https://ok.ru/videoembed/{@id}']
	]
);
```

http://s9e.github.io/TextFormatter/api/s9e/TextFormatter/Plugins/MediaEmbed/Configurator/Collections/CachedDefinitionCollection.html

Reply:
Oo that may be even better

Reply:
@JoshyPHP The goal is expanding it yeah. Thanks, expanding the collection looks much better.

Reply:
During development I keep site definitions in individual XML files. If you want to do that, there's another collection for it:
```php
use s9e\TextFormatter\Plugins\MediaEmbed\Configurator\Collections\XmlFileDefinitionCollection;
$path = '/path/to/xml/files/dir';
$collection = new XmlFileDefinitionCollection($path);
```

Reply:
That would be best @rxu for what I was hoping for... If people could just drop XML files for new sites into a folder within our ext that would be best

Reply:
Indeed. The only concern is that for the 'average user' preparing such an xml file may sound even more impossible than the site definition in PHP.
We'll have to give some simple instruction/boilerplates for this then.

Reply:
Also custom collection should expand the default one somehow.